NuRadioReco/modules/io/rno_g/readRNOGDataMattak.py

```diff
@@ -0,0 +1,569 @@
+      Returns
+      -------
+      
+      skip: bool
+         Returns True to skip/reject event, return False to keep/read event
+      """
+      if self._selectors is not None:
+         for selector in self._selectors:
+            if not selector(evtinfo):
+               self.logger.debug(f"Event {event_idx} (station {evtinfo.station}, run {evtinfo.run}, "
+                                 f"event number {evtinfo.eventNumber}) is skipped.")
+               self.__skipped += 1
+               return True
+      
+      return False
+   
+   
+   def get_event_information_dict(self, keys=["station", "run"]):
+      """ Return information of all events from the EventInfo
+      
+      This function is useful to make a pre-selection of events before actually reading them in combination with 
+      self.read_event().
+      
+      Parameters
+      ----------
+      
+      keys: list(str)
+         List of the information to receive from each event. Have to match the attributes (member variables)
+         of the mattak.Dataset.EventInfo class (examples are "station", "run", "triggerTime", "triggerType", "eventNumber", ...).
+         (Default: ["station", "run"])
+         
+      Returns
+      -------
+      
+      data: dict
+         Keys of the dict are the event indecies (as used in self.read_event(event_index)). The values are dictinaries 
+         them self containing the information specified with "keys" parameter.
+      """
+      
+      data = {}
+      n_prev = 0
+      for dataset in self._datasets:
+         dataset.setEntries((0, dataset.N()))
+         
+         for idx, evtinfo in enumerate(dataset.eventInfo()):  # returns a list
+      
+            event_idx = idx + n_prev  # event index accross all datasets combined 
+         
+            if self.filter_event(evtinfo, event_idx):
+               continue
+         
+            data[event_idx] = {key: getattr(evtinfo, key) for key in keys}
+         
+         n_prev += dataset.N()
+
+      return data
+   
+   
+   def get_event(self, event_info, waveforms):
+      """ Return a NuRadioReco event
+      
+      Parameters
+      ----------
+      
+      event_info: mattak.Dataset.EventInfo
+         The event info object for one event.
+         
+      waveforms: np.array(n_channel, n_samples)
+         Typically what dataset.wfs() returns (for one event!)
+
+      Returns
+      -------
+      
+      evt: NuRadioReco.framework.event
+      """
+      
+      evt = NuRadioReco.framework.event.Event(event_info.run, event_info.eventNumber)
+      station = NuRadioReco.framework.station.Station(event_info.station)
+      station.set_station_time(astropy.time.Time(event_info.triggerTime, format='unix'))
+
+      trigger = NuRadioReco.framework.trigger.Trigger(event_info.triggerType)
+      trigger.set_triggered()
+      trigger.set_trigger_time(event_info.triggerTime)
+      station.set_trigger(trigger)
+      
+      for channel_id, wf in enumerate(waveforms):
```

Comment:
@cozzyd might be able to give a answer. I would guess it does not make much of a difference

Reply:
Right now the way it's serialized all waveforms are stored contiguously on disk, I'm pretty sure, so there is no benefit to only reading some of them (they get decompressed etc. regardless). This could in principle be changed in the future.   But, if expensive calibration is to be done, there may a still benefit to not doing that to all waveforms if you don't need them?

Reply:
This can be a "nice to have" for the future